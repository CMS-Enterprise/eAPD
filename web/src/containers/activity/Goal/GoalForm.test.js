import { shallow } from 'enzyme';
import React from 'react';

import {
  setGoalDescription,
  setGoalObjective
} from '../../../actions/editActivity';

jest.spyOn(React, 'useContext');
React.useContext.mockImplementation(() => ({
  index: 'activity index'
}));

const { plain: GoalForm, mapDispatchToProps } = require('./GoalForm');

describe('the GoalForm component', () => {
  const props = {
    index: 1,
    item: {
      description: 'goal description',
      key: 'goal key',
      objective: 'goal objective'
    },
    setDescription: jest.fn(),
    setObjective: jest.fn()
  };

  beforeEach(() => {
    props.setDescription.mockClear();
    props.setDescription.mockClear();
  });

  test('renders correctly', () => {
    const component = shallow(<GoalForm {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('it handles changing the goal description', () => {
    const component = shallow(<GoalForm {...props} />);
    component
      .findWhere(c => c.prop('name') === 'name')
      .simulate('change', { target: { value: 'new name' } });

    expect(props.setDescription).toHaveBeenCalledWith(
      'activity index',
      1,
      'new name'
    );
  });

  it('it handles changing the goal objective', () => {
    const component = shallow(<GoalForm {...props} />);
    component
      .findWhere(c => c.prop('name') === 'milestones')
      .simulate('change', { target: { value: 'new objective' } });

    expect(props.setObjective).toHaveBeenCalledWith(
      'activity index',
      1,
      'new objective'
    );
  });

  it('maps dispatch actions to props', () => {
    expect(mapDispatchToProps).toEqual({
      setDescription: setGoalDescription,
      setObjective: setGoalObjective
    });
  });
});
