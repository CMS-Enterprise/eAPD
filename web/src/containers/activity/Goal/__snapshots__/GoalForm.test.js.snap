// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the GoalForm component renders correctly 1`] = `
<Fragment>
  <TextField
    autoFocus="true"
    className="data-entry-box"
    hint="Examples of goals include: improved health outcomes for benficiaries, transformation of public health services, enabling implementation of other initiatives, etc."
    label="Goal name"
    name="name"
    type="text"
    value="goal description"
  />
  <TextArea
    className="data-entry-box"
    hint="Explain how the state will make measurable progress towards the activity goal."
    label="Benchmarks"
    name="milestones"
    rows={6}
    value="goal objective"
  />
</Fragment>
`;
